Documentation/test-targets-usage.md

```diff
@@ -0,0 +1,108 @@
+Tests can be compiled and run with .NET Native by specifying the `UseDotNetNativeToolchain` property.
+
+```
+msbuild /t:BuildAndTest /p:TestTFM=netcore50aot /p:TestNugetRuntimeId=win10-x64-aot /p:UseDotNetNativeToolchain=true Microsoft.CSharp.Tests.csproj
+```
+
+#### Build and run tests against packages
+
+By default, the test project's direct dependencies (i.e. assemblies for the projects to which the test project has a ProjectReference, such as the corresponding product project) are copied from each referenced project's `$(TargetPath)`.  The `BuildTestsAgainstPackages` property provides the option to instead copy these references from a packages directory.
+```
+msbuild /t:BuildAndTest /p:BuildTestsAgainstPackages=true /p:"PackagesDrops=C:\Repos\corefx\bin\packages\Debug" System.Collections.Immutable.Tests.csproj
+```
+
+[Building Tests Against Packages](https://github.com/dotnet/corefx/blob/master/Documentation/building/build-tests-against-packages.md) has more information about doing so in CoreFX.
+
+#### Run code coverage tests
+
+Use the `Coverage` property to run code coverage tests.
+
+```
+msbuild /t:Test /p:Coverage=true System.Collections.Immutable.Tests.csproj
+```
+
+#### Run performance tests
+
+Use the `Performance` property to run performance tests.
+
+```
+msbuild /t:Test /p:Performance=true System.Collections.Immutable.Tests.csproj
+```
+
+#### Archive tests for running remotely
+
+Use the `ArchiveTests` property to archive tests.  This is typically used to prepare for running the tests in Helix.
+
+```
+msbuild /t:Test /p:ArchiveTests=true System.Collections.Immutable.Tests.csproj
+```
+
+The [CoreFX official build](https://github.com/dotnet/corefx/blob/021c590e6cb166c4bfc62b2c9966e317c37c1ed6/buildpipeline/DotNet-CoreFx-Trusted-Windows-Build-Test.json#L178-L179) provides an example of this usage.
+
+#### Run tests in Helix
+
+The `CloudBuild` target can be used to queue a test run in Helix.
+
+Here is an example based on [the CoreFX official build](https://github.com/dotnet/corefx/blob/021c590e6cb166c4bfc62b2c9966e317c37c1ed6/buildpipeline/DotNet-CoreFx-Trusted-Windows-Build-Test.json#L215-L216):
+```
+msbuild src\post.builds /t:CloudBuild /p:"BuildTestsAgainstPackages=true" /p:"ConfigurationGroup=Debug" /p:"ArchiveTests=true" /p:"EnableCloudTest=true" /p:"BuildMoniker=[build_moniker]" /p:"TargetQueue=Windows.10.Amd64" /p:"TestProduct=corefx" /p:"Branch=master" /p:"CloudDropAccountName=[cloud_drop_account_name]" /p:"CloudResultsAccountName=[cloud_results_account_name]" /p:"CloudDropAccessToken=[cloud_drop_access_token]" /p:"CloudResultsAccessToken=[cloud_results_access_token]" /p:"Outerloop=true" /p:"FilterToOSGroup=Windows_NT" /p:"TimeoutInSeconds=1200" /p:"TargetOS=Windows_NT" /p:"HelixApiAccessKey=[helix_key]" /p:"CloudTestContainerSuffix=windows10ilc" /p:"OverwriteOnUpload=true" /p:"TargetsWindows=true" /flp:"LogFile=pushtohelix-diag.log;Verbosity=diagnostic" /p:HelixApiEndpoint=https://helix.dot.net/api/2016-06-28/jobs
```

Comment:
I think I would avoid copying the command line here as you already provided a link which is good. Having the command line will cause people to be tempted to try and copy/paste it.